lock_api/src/mutex.rs

```diff
@@ -780,10 +788,12 @@ impl<R: RawMutex, T: ?Sized> Drop for ArcMutexGuard<R, T> {
 /// could introduce soundness issues if the locked object is modified by another
 /// thread.
 #[must_use = "if unused the Mutex will immediately unlock"]
-pub struct MappedMutexGuard<'a, R: RawMutex, T: ?Sized> {
+pub struct MappedMutexGuard<'a, R: RawMutex, T: ?Sized + 'a> {
```

Comment:
I don't think `T` has to live as long as `'a`, we can remove the `'a` bound on `T`.

Only `R` needs the bound because we have a `&'a R`.

Reply:
It definitely doesn't need the `'a` bound on the struct itself, and I think you're right that it isn't needed elsewhere either.